functional-test/src/test/groovy/org/rundeck/util/gui/pages/jobs/JobShowPage.groovy

```diff
@@ -48,6 +48,8 @@ class JobShowPage extends BasePage implements ActivityListTrait {
     By jobLinkTitleBy = By.xpath("//a[contains(@class, 'job-header-link')]")
     By autocompleteJobStepDefinitionBy = By.cssSelector("#wfitem_0 > span > div > div > span > span > span.text-success")
     By runFormBy = By.cssSelector("#execDiv #exec_options_form #formbuttons #execFormRunButton")
+    By runJobButtonBy = By.cssSelector(".btn.btn-success.btn-simple.btn-hover.btn-xs.act_execute_job")
+    By executionStateLabelBy = By.id("executionStateDisplayLabel")
```

Comment:
@mrdubr  before using it in my test I wanted to make sure if I am implementing it correctly?  Do I need to use these selector with the helper method clickRunJobButton and waitForExecutionState

Reply:
IIRC the ticket AC, your tests are not related to the JobShow page, thus I'm not sure you'd need this.
It's much simpler (and more maintainable) to setup (prepare) data for your tests using backend API.
There are numerous tests that create jobs and wait for executions to complete.
I'd recommend following that pattern.

Reply:
> IIRC the ticket AC, your tests are not related to the JobShow page, thus I'm not sure you'd need this. It's much simpler (and more maintainable) to setup (prepare) data for your tests using backend API. There are numerous tests that create jobs and wait for executions to complete. I'd recommend following that pattern.

I tried to refactor the tests by looking other tests. Though it's failing but I am keep working on it. It  doesn't look complicated but I may be missing something.

Reply:
Feel free to zoom me if you need some guidance on this